src/envelope.rs

```diff
@@ -39,14 +40,59 @@ const NONCE_LEN: usize = 32;
 pub(crate) struct Envelope<D: Hash> {
     nonce: Vec<u8>,
     ciphertext: Vec<u8>,
+    auth_data: Vec<u8>,
     hmac: GenericArray<u8, <D as Digest>::OutputSize>,
 }
 
+pub(crate) struct OpenedEnvelopeECF {
+    pub(crate) credentials_map: HashMap<CredentialType, Vec<u8>>,
```

Comment:
I keep thinking this `HashMap` is an overkill that actually subsumes a `BitMap` indexation into a `Vec`. There are some really tiny bitmap crates out there, here's my favorite:
https://github.com/bodil/bitmaps

Reply:
Perhaps we can leave this to a future change. Created #58 to track this